core/src/apps/bitcoin/get_public_key.py

```diff
@@ -21,7 +21,9 @@ async def get_public_key(ctx: wire.Context, msg: GetPublicKey) -> PublicKey:
     node = keychain.derive(msg.address_n)
 
     if (
-        script_type in [InputScriptType.SPENDADDRESS, InputScriptType.SPENDMULTISIG]
+        script_type in (InputScriptType.SPENDADDRESS, InputScriptType.SPENDMULTISIG)
+        or coin.decred
+        and script_type in (InputScriptType.SPENDSSRTX, InputScriptType.SPENDSSGEN)
         and coin.xpub_magic is not None
     ):
```

Comment:
My python isn't good enough. Does this still make it to the last condition if not coin.decred?

Reply:
Just put brackets around - so it is `or (coin.decred and script_type in (InputScriptType.SPENDSSRTX, InputScriptType.SPENDSSGEN))`